sonar-php-plugin/src/test/java/org/sonar/plugins/php/phpunit/CompatibilityImportersFactoryTest.java

```diff
@@ -0,0 +1,142 @@
+
+public class CompatibilityImportersFactoryTest {
+
+  private static final String BASE_DIR = "/org/sonar/plugins/php/phpunit/sensor/src/";
+  private static final Class[] LEGACY_IMPORTER_CLASSES = {
+    TestResultImporter.class,
+    CoverageResultImporter.class,
+    ItCoverageResultImporter.class,
+    OverallCoverageResultImporter.class
+  };
+  private static final SonarRuntime SONAR_QUBE_6_2 = SonarRuntimeImpl.forSonarQube(Version.create(6, 2), SonarQubeSide.SCANNER);
+  private static final SonarRuntime SONAR_QUBE_6_1 = SonarRuntimeImpl.forSonarQube(Version.create(6, 1), SonarQubeSide.SCANNER);
+  private static final String COVERAGES_KEY = PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATHS_KEY;
+  private static final String COVERAGE_KEY = PhpPlugin.PHPUNIT_COVERAGE_REPORT_PATH_KEY;
+  private static final String OVERALL_COVERAGE_KEY = PhpPlugin.PHPUNIT_OVERALL_COVERAGE_REPORT_PATH_KEY;
+  private static final String IT_COVERAGE_KEY = PhpPlugin.PHPUNIT_IT_COVERAGE_REPORT_PATH_KEY;
+
+  private SensorContextTester context;
+  private CompatibilityImportersFactory importersFactory;
+
+  @Before
+  public void setUp() throws Exception {
+    context = SensorContextTester.create(TestUtils.getResource(BASE_DIR));
+    importersFactory = new CompatibilityImportersFactory(context);
+  }
+
+  @Test
+  public void should_create_legacy_coverage_importers_before_6_2() throws Exception {
+    context.setRuntime(SONAR_QUBE_6_1);
+    final List<Class> importerClasses = importerClasses(importersFactory.createImporters());
+    assertThat(importerClasses).containsExactly(LEGACY_IMPORTER_CLASSES);
+  }
+
+  @Test
+  public void should_create_test_result_and_multi_coverage_importer_starting_from_6_2() throws Exception {
+    context.setRuntime(SONAR_QUBE_6_2);
+    context.settings().setProperty(COVERAGES_KEY, "coverage report");
+    final List<Class> importerClasses = importerClasses(importersFactory.createImporters());
+    assertThat(importerClasses).containsExactly(TestResultImporter.class, MultiPathImporter.class);
+  }
+
+  @Test
+  public void should_fallback_to_legacy_importers_if_multipath_coverage_property_not_used() throws Exception {
+    context.setRuntime(SONAR_QUBE_6_2);
+    final List<Class> importerClasses = importerClasses(importersFactory.createImporters());
+    assertThat(importerClasses).containsExactly(LEGACY_IMPORTER_CLASSES);
+  }
+
+  @Test
+  public void should_print_deprecation_warning_if_legacy_properties_used_from_6_2() throws Exception {
+    context.setRuntime(SONAR_QUBE_6_2);
+    context.settings().setProperty(COVERAGE_KEY, "coverage report");
+    assertThat(importersFactory.deprecationWarnings()).containsExactly(
+      String.format(DEPRECATION_WARNING_TEMPLATE, COVERAGE_KEY));
+    context.settings().setProperty(IT_COVERAGE_KEY, "integration coverage report");
+    context.settings().setProperty(OVERALL_COVERAGE_KEY, "overall coverage report");
+    assertThat(importersFactory.deprecationWarnings()).containsExactly(
```

Comment:
@inverno it's hard to read 2 groups of assertions within one method without even an empty line